### PR TITLE
Fix Bug : Prevent item crafting with Island Crystals.

### PR DESCRIPTION
**Pull request by DoctaEnkoda**

### CODE REVIEW

File: src/main/java/com/iridium/iridiumskyblock/listeners/ItemCraftListener.java [hunk 1]
```diff
@@ -37,4 +42,20 @@ public void monitorItemCraft(CraftItemEvent event) {
         island.ifPresent(value -> IridiumSkyblock.getInstance().getIslandManager().incrementMission(value, "CRAFT:" + material.name(), amount));
     }
 
+    @EventHandler(priority = EventPriority.LOW)
+    public void noUseCrystalCraft(CraftItemEvent event) {
```
Comment (MertUnverdi) [inline src/main/java/com/iridium/iridiumskyblock/listeners/ItemCraftListener.java:46]: we must use PreItemCraft event for this

File: src/main/java/com/iridium/iridiumskyblock/listeners/ItemCraftListener.java [hunk 2]
```diff
@@ -37,4 +42,20 @@ public void monitorItemCraft(CraftItemEvent event) {
...
+        for (ItemStack stack : event.getInventory().getMatrix()) {
+            if (stack != null && stack.getType().equals(configuration.islandCrystal.material.parseMaterial())) {
+                ItemMeta itemMeta = stack.getItemMeta();
+                if (itemMeta != null && itemMeta.getDisplayName().contains("Island Crystal")) {
+                    event.setCancelled(true);
+                    event.getInventory().setResult(air);
+                    return;
+                }
+            }
+        }
```
Comment (MertUnverdi) [inline src/main/java/com/iridium/iridiumskyblock/listeners/ItemCraftListener.java:58]:
```suggestion
        for (ItemStack itemStack : event.getInventory().getMatrix()) {
            if (itemStack != null && IridiumSkyblock.getInstance().getIslandManager().getIslandCrystals(itemStack) > 0) {
                    event.setCancelled(true);
                    event.getInventory().setResult(new ItemStack(Material.AIR));
            }
        }
```

File: src/main/java/com/iridium/iridiumskyblock/listeners/ItemCraftListener.java [hunk 3]
```diff
@@ -3,21 +3,26 @@
...
+import org.bukkit.inventory.meta.ItemMeta;
 
 import java.util.Arrays;
 import java.util.Objects;
 import java.util.Optional;
 
 public class ItemCraftListener implements Listener {
 
+    ItemStack air = new ItemStack(Material.AIR);
+
```
Comment (MertUnverdi) [inline src/main/java/com/iridium/iridiumskyblock/listeners/ItemCraftListener.java:25]:
```suggestion
```

File: src/main/java/com/iridium/iridiumskyblock/listeners/ItemCraftListener.java [hunk 4]
```diff
@@ -3,21 +3,26 @@
 import com.iridium.iridiumcore.dependencies.xseries.XMaterial;
 import com.iridium.iridiumskyblock.IridiumSkyblock;
 import com.iridium.iridiumskyblock.api.IridiumSkyblockAPI;
+import com.iridium.iridiumskyblock.configs.Configuration;
```
Comment (MertUnverdi) [inline src/main/java/com/iridium/iridiumskyblock/listeners/ItemCraftListener.java:6]:
```suggestion
```

File: src/main/java/com/iridium/iridiumskyblock/listeners/ItemCraftListener.java [hunk 5]
```diff
@@ -3,21 +3,26 @@
...
 import com.iridium.iridiumskyblock.database.Island;
 import com.iridium.iridiumskyblock.database.User;
+import org.bukkit.Material;
 import org.bukkit.entity.Player;
 import org.bukkit.event.EventHandler;
 import org.bukkit.event.EventPriority;
 import org.bukkit.event.Listener;
 import org.bukkit.event.inventory.CraftItemEvent;
 import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
```
Comment (MertUnverdi) [inline src/main/java/com/iridium/iridiumskyblock/listeners/ItemCraftListener.java:16]:
```suggestion
```

**Review by MertUnverdi — CHANGES_REQUESTED** [review]

**Comment by MertUnverdi** [comment]
did tested with PreItemCraft event ?

**Comment by DoctaEnkoda** [comment]
Yes I'm test, it work !

**Comment by DoctaEnkoda** [comment]
Island Crystals : 
image
Nether Star : 
image

File: src/main/java/com/iridium/iridiumskyblock/listeners/ItemCraftListener.java [hunk 6]
```diff
@@ -37,4 +39,13 @@ public void monitorItemCraft(CraftItemEvent event) {
         island.ifPresent(value -> IridiumSkyblock.getInstance().getIslandManager().incrementMission(value, "CRAFT:" + material.name(), amount));
     }
 
+    @EventHandler(priority = EventPriority.LOW)
+    public void noUseCrystalCraft(PrepareItemCraftEvent event) {
```
Comment (MertUnverdi) [inline src/main/java/com/iridium/iridiumskyblock/listeners/ItemCraftListener.java:43]: More better name :panda:
```suggestion
    public void onItemCraft(PrepareItemCraftEvent event) {
```

**Review by MertUnverdi — CHANGES_REQUESTED** [review]

**Review by MertUnverdi — APPROVED** [review]

**Review by MertUnverdi — APPROVED** [review]
idk what im doing -,-

**Review by dlsf — APPROVED** [review]

File: src/main/java/com/iridium/iridiumskyblock/listeners/ItemCraftListener.java [hunk 7]
```diff
@@ -37,4 +39,13 @@ public void monitorItemCraft(CraftItemEvent event) {
         island.ifPresent(value -> IridiumSkyblock.getInstance().getIslandManager().incrementMission(value, "CRAFT:" + material.name(), amount));
     }
 
+    @EventHandler(priority = EventPriority.LOW)
+    public void onItemCraft(PrepareItemCraftEvent event) {
+        for (ItemStack itemStack : event.getInventory().getMatrix()) {
+            if (itemStack != null && IridiumSkyblock.getInstance().getIslandManager().getIslandCrystals(itemStack) > 0) {
+                event.getInventory().setResult(new ItemStack(Material.AIR));
```
Comment (PeachesMLG) [inline src/main/java/com/iridium/iridiumskyblock/listeners/ItemCraftListener.java:46]: think we should set to null instead, also no need to check if itemStack != null we do that inside the getIslandCrystals method

**Review by PeachesMLG — CHANGES_REQUESTED** [review]

**Pull request merged**